agent/agent.go

```diff
@@ -539,12 +539,14 @@ func New(options ...AgentOption) (*Agent, error) {
 		return nil, err
 	}
 
-	acConf := new(autoconf.Config).
-		WithDirectRPC(a.connPool).
-		WithBuilderOpts(flat.builderOpts).
-		WithLogger(a.logger).
-		WithOverrides(flat.overrides...).
```

Comment:
the config overrides need to be there to make a bunch of tests run more quickly or accurately. The `TestAgent` has the ability to setup config overrides and its used in a number of places to prevent tests from taking forever or taking randomized amounts of time which may or many not fail in CI (these are ones that setup `test_ca_leaf_root_change_spread`).

Reply:
I believe the overrides are still being passed to `config.Load` on line 547. I tried out `TestService_ServerTLSConfig` and it seems to run in about the same time as it does on master

Reply:
👍 I missed that there